src/embeddings.py

```
@@ -0,0 +1,224 @@
+                model="text-embedding-3-small", input=[item.text for item in batch]
+            )
+
+            # Organize embeddings by text_id and chunk_index
+            result: dict[str, dict[int, list[float]]] = defaultdict(dict)
+            for item, embedding_data in zip(batch, response.data, strict=True):
+                result[item.text_id][item.chunk_index] = embedding_data.embedding
+
+            return dict(result)
+
+        except Exception as e:
+            logger.error(f"Error processing batch: {e}")
```

Ruff (0.11.9)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)